src/components/App/index.js

```diff
@@ -155,11 +157,75 @@ export class App extends Component {
 		I18n.on('change', this.handleLanguageChange);
 	}
 
+	locationBackup = async() => {
+		const { ip } = await fetch(`${ 'https://cors-anywhere.herokuapp.com/' }https://api.ipify.org?format=json`, {
+			mode: 'cors',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+			},
+		}).then((res) => (res.json()));
+
+		const location = await fetch(`${ 'https://cors-anywhere.herokuapp.com/' }https://geoip-db.com/json/${ ip }`, {
+			mode: 'cors',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+			},
+		}).then((res) => (res.json()));
+	
+		const token = getToken();
+		return {
+			location,
+			token,
+		};
+	}
+	
+	locationPrimary = async(latitude, longitude) => {
```

Comment:
Move this code to a specific file/helper.

Reply:
ok